plugins/tactics/src/Ide/Plugin/Tactic/Types.hs

```diff
@@ -65,7 +65,7 @@ instance Show Var where
 instance Show TCvSubst where
   show  = unsafeRender
 
-instance Show (LHsExpr GhcPs) where
+instance {-# OVERLAPPING #-} Show (LHsExpr GhcPs) where
```

Comment:
```
src\Ide\Plugin\Tactic.hs:310:15: error:
    • Overlapping instances for Show
                                  (GenLocated SrcSpan (HsExpr GhcPs))
        arising from a use of ‘traceMX’
      Matching instances:
        instance Outputable.Outputable a => Show (GenLocated SrcSpan a)
          -- Defined in ‘ghcide-0.4.0:Development.IDE.GHC.Orphans’
        instance Show (LHsExpr GhcPs)
          -- Defined at src\Ide\Plugin\Tactic\Types.hs:68:10
    • In the expression: traceMX "solns"
      In a stmt of a 'do' block: traceMX "solns" $ rtr_other_solns rtr
      In the expression:
        do traceMX "solns" $ rtr_other_solns rtr
           let g = graft (RealSrcSpan span) $ rtr_extract rtr
               response = transform dflags (clientCapabilities lf) uri g pm
           pure
             $ case response of
                 Right res -> (Right Null, Just ...)
                 Left err
                   -> (Left $ ResponseError InternalError (T.pack err) Nothing,
                       Nothing)
    |
310 |               traceMX "solns" $ rtr_other_solns rtr
    |               ^^^^^^^^^^^^^^^
```

Reply:
I think you can just delete this?

Reply:
I tried to delete the ghcide instance and it thrower an error, will doublechek

Reply:
I mean the HLS instance.

Reply:
removing it it throws:
```
src\Ide\Plugin\Tactic\Types.hs:368:14: error:
    • No instance for (Show (LHsExpr GhcPs))
        arising from the second field of ‘RunTacticResults’
          (type ‘LHsExpr GhcPs’)
      Possible fix:
        use a standalone 'deriving instance' declaration,
          so you can specify the instance context yourself
    • When deriving the instance for (Show RunTacticResults)
    |
368 |   } deriving Show
    |              ^^^^
```
It uses the instance in other context.

----------------------

I dont fullt understand why they overlap

For 
`Show (GenLocated SrcSpan (HsExpr GhcPs))`

`instance Outputable.Outputable a => Show (GenLocated SrcSpan a)`
and
`instance Show (LHsExpr GhcPs)`

Reply:
I guess we could make hls use `Outputable` for `(LHsExpr GhcPs)` instead `Show` via `showSDoc flags . ppr` or `showSDocUnsafe . ppr`

Reply:
> removing it it throws:

Is the `Orphans` module imported over there?

Reply:
It is not, but i am afraid importing `Development.IDE.GHC.Orphans` in `Ide.Plugin.Tactic.Types` (after exposing the module in ghcide, cause it is a hidden module now) does not remove the error `No instance for (Show (LHsExpr GhcPs))...`

Reply:
Very confusing, there is definitely an instance for `Outputable (HsExpr p)`. Try importing `HsExpr`?

Reply:
> Very confusing

@wz1000 my bad, i was not testing the correct branch, using the ghcide version of this pr and making public `Development.IDE.GHC.Orphans` to include it in `Ide.Plugin.Tactic.Types` and removing the instance for `LHsExpr GhcPs`, makes it work

it would need a pr in ghcide to expose Orphans, do you think we should take that right way or merging this as-is with a TODO comment to not block the hls incoming release?

Reply:
pr opened (with the branch in the main repo just in case): https://github.com/haskell/ghcide/pull/894

Reply:
I'm sure the instances are exported through some other module? You just have to import that.

Reply:
Ok will check what module reexports them

Reply:
It turns out that several modules export those instances, as @wz1000  anticipated.
Of all of them i have chose `Development.IDE ()`, the more generic, cause import a specific module (for example Development.Ide.Core.Sahe) only for instances does not look sensible.

However i would use `Development.IDE.GHC.Orphans` if it is visible cause the intent of the import is clearer, so i would keep the pr exporting it opened. We will can use it here afterwards in hls.

Reply:
Agreed - importing Orphans is much clearer.